src/backend/category.ml

```diff
@@ -0,0 +1,35 @@
+
+(* This file is free software, part of FTW. See file "LICENSE" for more information *)
+
+(* Routes *)
+(* ************************************************************************* *)
+    
+let rec routes router =
+  router
+  |> Router.get "/api/categories" category_list
+    ~tags:["category"]
+    ~summary:"List existing categories"
+    ~responses:[
+      "200", Types.obj @@ Spec.make_response_object ()
+        ~description:"Succesful operation"
+        ~content:[
+          "application/json",
+          Spec.make_media_type_object () ~schema:Types.(ref CategoryList.ref);
+        ];
+    ]
+
+
+(* Competition query *)
+(* ************************************************************************* *)
+
+and category_list =
+  Api.get
+    ~to_yojson:Types.CategoryList.to_yojson
+    (fun _req _st ->
+      let categories = Types.Category.all in
+      let res : Types.CategoryList.t = { categories; } in
+      Ok res
+    )
```

Comment:
I'm not sure what's the use of an API call to return the list of categories: the type of categories already defines that list, so it seems a bit redundant and wasteful to have an API call to return something that is static and known at compile-time for both the backend and frontend.

Reply:
The frontend will **have* to statically know about the different categories, if onyl to properly print and display them for each competition. Said otherwise, if at any point the frontend does not statically know about all of the categories, then it is a bug. I agree that it means additional work to ensure that the frontend and backend are in sync, but it's unavoidable since the two have separate code.